src/librustdoc/visit_ast.rs

```diff
@@ -506,11 +523,35 @@ impl<'a, 'tcx> RustdocVisitor<'a, 'tcx> {
             | hir::ItemKind::TraitAlias(..) => {
                 self.add_to_current_mod(item, renamed, import_id);
             }
-            hir::ItemKind::Const(..) => {
+            hir::ItemKind::Const(_, ty, _, _) => {
                 // Underscore constants do not correspond to a nameable item and
                 // so are never useful in documentation.
                 if get_name() != kw::Underscore {
-                    self.add_to_current_mod(item, renamed, import_id);
+                    if self.cx.cache.document_tests
+                        && let TyKind::Path(rustc_hir::QPath::Resolved(_, path)) = ty.kind
+                    {
+                        let path: String = path
+                            .segments
+                            .iter()
+                            .map(|s| {
+                                if s.ident.name == kw::PathRoot {
+                                    ""
+                                } else {
+                                    s.ident.name.as_str()
+                                }
+                            })
+                            .intersperse("::")
+                            .collect();
+                        if path == "test::TestDescAndFn" {
```

Comment:
```suggestion
                        let path_names = path
                            .segments
                            .iter()
                            .map(|s| {
                                s.ident.name
                            });
                        if path_names.eq([sym::test, sym::TestDescAndFn]) {
```

I think, to follow this suggestion, you'd need to add TestDescAndFn to <https://doc.rust-lang.org/nightly/nightly-rustc/src/rustc_span/symbol.rs.html#169>